lib/puppet/pops/parser/egrammar.ra

```diff
@@ -655,7 +655,8 @@ classname
   : NAME
   | WORD
   | CLASSREF
-  | CLASS { error val[0], "'class' is not a valid classname" }
+  | CLASS { error val[0], "'class' keyword not allowed at this location" }
+  | STRING { error val[0], "a string was found where it is not accepted" }
```

Comment:
my suggestion was to use "A quoted string is not acceptable as a name" - for that it is clear that it is expecting a name but got a quoted string (as opposed to a bare word which is also a string).